src/CommentSold/Api/Resources/Request/Account/CreateShopRequest.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CommentSold\Api\Resources\Request\Account;
+
+use CommentSold\Api\Resources\Request\AbstractRequest;
+
+class CreateShopRequest extends AbstractRequest
+{
+    public string $shop_id;
+    public string $shop_name;
+    public string $company_name;
+    public string $company_phone;
+    public string $street_address;
+    public string $city;
+    public string $state;
+    public string $country_code;
+    public string $zip;
+    public string $first_name;
+    public string $last_name;
+    public string $email;
+    public string $webhook_url;
+    public ?string $account_type;
+    public ?bool $no_customer_emails;
+
+    public function __construct(array $payload)
+    {
+        $this->shop_id            = $payload['shop_id'];
+        $this->shop_name          = $payload['shop_name'];
+        $this->company_name       = $payload['company_name'];
+        $this->company_phone      = $payload['company_phone'];
+        $this->street_address     = $payload['street_address'];
+        $this->city               = $payload['city'];
+        $this->state              = $payload['state'];
+        $this->country_code       = $payload['country_code'];
+        $this->zip                = $payload['zip'];
+        $this->first_name         = $payload['first_name'];
+        $this->last_name          = $payload['last_name'];
+        $this->email              = $payload['email'];
+        $this->webhook_url        = $payload['webhook_url'];
+        $this->account_type       = $payload['account_type'] ?? null;
+        $this->no_customer_emails = $payload['no_customer_emails'] ?? null;
+    }
+}
```

Comment:
Nits:

* I think camelCase is much more common in the PHP ecosystem.
* I don't really see any benefit to passing through an array first. In the end, you have 13 required parameters and 2 optional ones, whether you wrap them in an array or not. If anything, the array wrapper just obscures what those parameters are. 
* I prefer readonly parameters, although I could see the argument for keeping a distributed SDK more flexible.

Reply:
> * I think camelCase is much more common in the PHP ecosystem.

I was keeping everything consistent with the API documentation so it was obvious what data to pass.

> * I don't really see any benefit to passing through an array first. In the end, you have 13 required parameters and 2 optional ones, whether you wrap them in an array or not. If anything, the array wrapper just obscures what those parameters are.

So make everything a parameter that is passed? I could do that I suppose. I'd have to make the (currently) internal objects exposed. So if you are creating a product you'd need to create an image object and then pass that in etc.

Reply:
I probably would. There's not really any validation otherwise.